Patch-Source: https://github.com/FreeRDP/FreeRDP/commit/c97fb38a4fa2c638a6e5755d3f5b4855e3ab9f8b
---
From c97fb38a4fa2c638a6e5755d3f5b4855e3ab9f8b Mon Sep 17 00:00:00 2001
From: akallabeth <akallabeth@posteo.net>
Date: Fri, 11 Apr 2025 08:17:01 +0200
Subject: [PATCH] [winpr,utils] ignore _Unwind_Backtrace return

See https://github.com/FreeRDP/FreeRDP/issues/11490
there is no consensus on what the return value should be, so warn but
ignore it. The context might contain a NULL stack trace, but that is
better than not to show anything.
---
 winpr/libwinpr/utils/unwind/debug.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/winpr/libwinpr/utils/unwind/debug.c b/winpr/libwinpr/utils/unwind/debug.c
index 585f21d7c538..d30b377aeb91 100644
--- a/winpr/libwinpr/utils/unwind/debug.c
+++ b/winpr/libwinpr/utils/unwind/debug.c
@@ -139,10 +139,14 @@ void* winpr_unwind_backtrace(DWORD size)
 	rc = _Unwind_Backtrace(unwind_backtrace_callback, ctx);
 	if (rc != _URC_END_OF_STACK)
 	{
+		/* https://github.com/FreeRDP/FreeRDP/issues/11490
+		 *
+		 * there seems to be no consensus on what to return from this function.
+		 * so we just warn about unexpected return codes and return the context regardless.
+		 */
 		char buffer[64] = { 0 };
-		WLog_ERR(TAG, "_Unwind_Backtrace failed with %s",
-		         unwind_reason_str_buffer(rc, buffer, sizeof(buffer)));
-		goto fail;
+		WLog_WARN(TAG, "_Unwind_Backtrace failed with %s",
+		          unwind_reason_str_buffer(rc, buffer, sizeof(buffer)));
 	}
 
 	return ctx;
